### ISSUE TITLE
Automatic handling of refresh vs access tokens

### ISSUE THREAD

**Issue by vmiklos** [pR0Ps/strava-backup#3]
Thanks for this great tool and also for strava-tokengen! One difficulty I found is that tokengen gives me a refresh token, while backup wants an access token. Currently I work this around in a bash script, see <https://github.com/vmiklos/vmexam/blob/master/strava/backup>. But it would be much better, if backup would accept a refresh token, and it would manage the creation (and if necessary, update) of the the access token.

I'm fine with looking into this myself, in case the feature (in theory) would be accepted in the form or a PR. What's your opinion about this?

**Comment by pR0Ps** [comment]
I built this before Strava updated their authentication process so all that was required was a single token. As of now, I think it makes the most sense to require a refresh token and exchange it for an access token before starting the script like you've done with your bash script.

I've actually already done ~90% of the work for this, but never got around to finishing it off so thanks for reminding me. I'll see if I can get it in a workable state and push it up soon.

**Comment by vmiklos** [comment]
Thanks!
